### PR TITLE
Added Befunge Testing

### PR DESCRIPTION
**Pull request by stuin**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][renegade-coder]! 
For simplicity, please make sure that your pull request includes one and only one contribution.

Please fill _one_ of the sections below as applicable.
Please also add any other relevant information to the Notes section at the bottom.
You may delete or just ignore any other sections.
For more information please refer to our [contributing documentation][contributing]

## I Am Adding a New Code Snippet in an Existing Language

- [ ] I fixed #your-issue-number-here
- [x] I named the pull request using `Add {PROJECT} in {LANGUAGE}` format
- [x] I created/updated the language README (see [contributing documentation][contributing-readme])
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][contributing-plagiarism])

  
## I Am Adding a New Code Snippet in a New Language

- [ ] I fixed #your-issue-number-here
- [ ] I named the pull request using `Add {PROJECT} in {LANGUAGE}` format
- [ ] I added a `testinfo.yml` files (see [contributing documentation][contributing-new-language])
  - [ ] I used an officially supported docker image or one that I personally trust
- [ ] I created/updated the language README (see [contributing documentation][contributing-readme])
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][contributing-plagiarism])
  

## I Am Adding New Tests for a Project

- [ ] I fixed #your-issue-number-here
- [ ] I named the pull request using `Add {PROJECT} tests` format
- [ ] I verified the project exists on the [Sample Programs Project List][sample-programs-project-list]
- [ ] I added tests for every test case in the Testing table of the project description for the given project. (See [contributing documentation][contributing-tests-in-detail])

## I Am Adding New Tests for a Language

- [ ] I fixed #your-issue-number-here
- [ ] I named the pull request using `Add {LANGUAGE} tests` format\
- [x] I added a `testinfo.yml` files (see [contributing documentation][contributing-new-language])
  - [x] I used an officially supported docker image or one that I personally trust
- [x] I verified all tests are passing

  
## I Am Modifying an Existing Code Snippet or Existing Tests

- [ ] I fixed #your-issue-number-here
- [ ] I have read about modifying existing code snippets or tests in the [contributing documentation][contributing-modifications]
- [ ] I have explained why this modification is _necessary_ either in the issue linked above or in the notes below
- [ ] I named the pull request using `Modify {PROJECT} {tests} in {LANGUAGE}` format

## Other Notes

Add notes here...

[renegade-coder]: https://therenegadecoder.com/
[contributing-plagiarism]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[contributing-new-project]: https://github.com/TheRenegadeCoder/sample-programs/blob/contributing/.github/CONTRIBUTING.md#requirements-for-a-new-project
[contributing-readme]: https://github.com/TheRenegadeCoder/sample-programs/blob/contributing/.github/CONTRIBUTING.md#create-readmes
[contributing-tests-in-detail]: https://github.com/TheRenegadeCoder/sample-programs/blob/contributing/.github/CONTRIBUTING.md#tests-in-detail
[contributing]: ../CONTRIBUTING.md
[sample-programs-project-list]: https://sample-programs.therenegadecoder.com/projects/
[contributing-modifications]: https://github.com/TheRenegadeCoder/sample-programs/blob/contributing/.github/CONTRIBUTING.md#modifying-existing-code-snippets

### CODE REVIEW

File: archive/b/befunge/README.md [hunk 1]
```diff
@@ -4,8 +4,9 @@ Welcome to Sample Programs in Befunge!
 
 ## Sample Programs
 
+- Baklava in Befunge
```
Comment (auroq) [inline archive/b/befunge/README.md:7]: Could you create an issue for to add an article for this and link it here?

**Review by auroq — CHANGES_REQUESTED** [review]
This looks pretty good to me (not that I can read the code).

However, could you change the extension to be to match the other snippets in that directory.

Also, would you be willing to add our `testinfo.yml` for this language so that we can get it tested?

https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#tests-in-detail

**Comment by stuin** [comment]
Tests mostly work, just that befunge can very easily output lots of extra spacing and invisible characters. I will see if I can fix all the outputs soon, but that will involve all of the existing befunge files.

**Comment by auroq** [comment]
> Tests mostly work, just that befunge can very easily output lots of extra spacing and invisible characters. I will see if I can fix all the outputs soon, but that will involve all of the existing befunge files.

Thank you for taking that on! We're trying to catch up on tech debt of untested snippets.
Feel free to modify existing snippets as needed to make them match our testing specs.

**Comment by stuin** [comment]
Fizz-Buzz was failing because of extra spaces added by the interpreter, so hopefully this other simpler container will work better.

**Comment by auroq** [comment]
> Fizz-Buzz was failing because of extra spaces added by the interpreter, so hopefully this other simpler container will work better.

I don't love the idea of ignoring whitespace in our tests, especially given the generally simple nature of the projects we support.

That said, I've played around with esoteric languages before, so I know that whitespace can be very difficult especially if added by the interpreter.

I just added a commit to this PR that I think is a hybrid approach. We have one test ignore the whitespace and a second test that counts lines to ensure that code snippets are doing some semblance of proper formatting.

Trying not to set a precedence of trimming formatting, but we did just have to make a similar change here due to a language interpreter: https://github.com/TheRenegadeCoder/sample-programs/pull/2223#pullrequestreview-509736677

@jrg94 would you like to weigh in. Any concerns?

**Comment by jrg94** [comment]
Wait, so the new tests basically remove all whitespace and check for content? Then, the second test makes sure the line count is correct? I think that's fine. 

I've had to do similar things for other personal projects, and I feel like more flexible testing will just save us time in the long term—especially because we can't visually see the white spaces.

That said, I agree! We should continue to be strict about whitespace where it makes sense.

**Comment by stuin** [comment]
I am pretty sure all the requested changes have been made. Is there anything I missed?

**Review by jrg94 — APPROVED** [review]
Looks good to me! 👍 

I would be careful with this image for future testing. I tried using a similar image for Brainfuck, and it doesn't accept input. I could be wrong though!

**Pull request merged**